scio-core/src/main/java/com/spotify/scio/transforms/BatchedBaseAsyncLookupDoFn.java

```diff
@@ -237,6 +237,15 @@ private FutureType handleOutput(FutureType future, List<Input> batchInput, UUID
                   pair -> {
                     final String id = pair.getLeft();
                     final Output output = pair.getRight();
+
+                    if(!inputs.containsKey(id)) {
```

Comment:
It took me a bit of time to realize my mistake. But basically if there is a miss match between the ids extracted from the input and the ids returned by the gRPC endpoint we only get an `IllegalStateException` with the following message `Expected requestCount == responseCount, but %s != %s`
throwing an exception inside the handleOutput is useless since we really don't operate on the futures after the `handleOutput` callbacks are added so i opted for adding and error log instead. 

If you have any better suggestions let me know :)

Reply:
makes sense .. i added the remainder part of the code in the else so it will only be executed if the inputs contain the key